mobify/astro-scaffold
# add deeplinking code and refactor to scaffold

I had added some code to AppDelegate in Sandbox to make deeplinking work but I didn't add it to the scaffold project. The user will still need to define their custom url scheme for deeplinking to work (unless Shawn puts that into his generator script).

Status: **Opened for review**

Reviewers: @jansepar @jeremywiebe @jvaill @MikeKlemarewski 
JIRA: **https://mobify.atlassian.net/browse/HYB-417**
Linked PRs: **https://github.com/mobify/astro/pull/183 https://github.com/mobify/astro/pull/201**
## Changes
- Copy and paste some boilerplate deeplinking code into the scaffold project.
- Setup getStartUri and receievedDeepLink in app.js so there is some boilerplate there out of the box
- config changes to correct how we reference app.js. @MikeKlemarewski believes that he may have left something in a broken state here and these project changes seem to fix that.
## How to test-drive this PR
- bring up scaffold project in simulator
- enter a custom Url scheme under "URL types" (instructions: https://github.com/mobify/astro/blob/develop/guides/deeplinks.md)
- go to Safari in the simulator and enter your url to open your app. If your url contains a webpage, it will open that page in a webview (via both launch and already running states). For example, enter <your_url_scheme>://http:///www.reddit.com and the app will launch and navigate to reddit.

IMPORTANT NOTE: you need all _three_ slashes after http if you are entering your url in safari. If you enter your url in notepad or calendar this is not required. If you don't have 3 slashes safari strips away the ":" in "http://" and the url is no longer valid. @jeremywiebe believes this is safari trying to be smart about valid/invalid urls. This shouldn't happen in the wild for two reasons:
1) people shouldn't be entering deep link urls in the safari's browser bar regularly
2) embedded deep link urls should be properly url encoded
## TODOS:
- [ ] Ensure changes to docs work well by changing them to Readme.io
